controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1312,7 +1313,13 @@ func (r *BareMetalHostReconciler) updateEventHandler(e event.UpdateEvent) bool {
 // SetupWithManager reigsters the reconciler to be run by the manager
 func (r *BareMetalHostReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
-	maxConcurrentReconciles := 3
+	maxConcurrentReconciles := runtime.NumCPU()
+	if maxConcurrentReconciles > 8 {
```

Comment:
Instead of using an hard-coded value, I was thinking that it could be useful to set as the upper limit something like `min(PROVISIONIG_LIMIT/2.5, 1)`, just to enforce the ratio between the two values _in case_ the user will set a particularly low value for `PROVISIONING_LIMIT`.

Reply:
> Is there a plan to keep this configurable as an option (via env)?

It is still configurable.

Reply:
> Instead of using an hard-coded value, I was thinking that it could be useful to set as the upper limit something like `min(PROVISIONIG_LIMIT/2.5, 1)`, just to enforce the ratio between the two values _in case_ the user will set a particularly low value for `PROVISIONING_LIMIT`.

I can't imagine anyone setting a smaller limit except on very small hardware, in which case the CPU count should scale the default appropriately for us. I can imagine people setting a bigger limit, but I don't think we want to increase the default concurrency in that case. I don't think we should tie these two settings together because ultimately they limit different things.

Reply:
It's not about tying them together, it's just about the currently proposed value for the _default_  upper limit of `maxConcurrentReconciles` - and I think there was an error in my previous comment, as I meant something like:

`maxConcurrentReconciles = min(PROVISIONING_LIMIT/2.5, runtime.NumCPU())`

which for a default PROVISIONING_LIMIT value of 20 will limit maxConcurrentReconciles to 8 on a system with more than 8 cores.

Given that the current proposal is to define the _default_ value based on the number of available CPU, the basic idea is to  try using all the available cpus as much possible - even though the benefits could be marginal (in general I'd expect that BMO will run within a container, which could have already resources and/or limits set), and at the same time trying to honor the fact that `maxConcurrentReconciles << PROVISIONING_LIMIT`